e2e/main-flow.test.ts

```diff
@@ -1038,3 +1014,55 @@ test.describe('Favorite Airlines Dialog', () => {
 		await expect(page.getByText('No airlines found.')).toBeVisible();
 	});
 });
+
+test.describe('Filter Regions', () => {
+	test.beforeEach(async ({ page }) => {
+		await preparePage(page, true);
+	});
+
+	test('should filter airlines by region', async ({ page }) => {
+		// Get initial number of rows
+		const initialRows = await page.getByRole('row').count();
+		expect(initialRows).toBeGreaterThan(0);
+
+		// Deselect all regions except one
+		const europeCheckbox = page.getByRole('button', { name: 'Europe' });
+		await page.getByText('Clear All').click();
+		await europeCheckbox.click();
+
+		// Wait for the table to be shown after filtering
+		await expect(page.getByRole('table')).toBeVisible();
+
+		// Get filtered number of rows and verify it's less than initial
+		const filteredRows = await page.getByRole('row').count();
+		expect(filteredRows).toBeLessThan(initialRows);
+
+		// Verify all visible airlines are from Europe
+		const regionCells = page.getByRole('cell', { name: 'Europe' });
+		const regionCount = await regionCells.count();
+		expect(regionCount).toBeGreaterThan(0);
+		expect(regionCount).toBe(filteredRows - 1);
+	});
+
+	test('should persist selected regions across page reloads', async ({ page }) => {
+		// Select Europe region
+		const europeButton = page.getByRole('button', { name: 'Europe' });
+		await page.getByText('Clear All').click();
+		await europeButton.click();
+
+		// Reload page
+		await page.reload();
+		await pageIsReady(page);
+
+		// Verify Europe region is still selected
+		await expect(europeButton).toHaveAttribute('data-active', 'true');
```

Comment:
You've added `data-selected` attribute for region toggle button, but looking for `data-active` in tests.